tests/cpp/test_multidevice_pipeline.cpp

```diff
@@ -307,14 +307,20 @@ INSTANTIATE_TEST_SUITE_P(
 enum class SchedulingMode {
   InterDeviceOnly,
   Manual,
+  ReductionScheduler,
+  Automatic,
 };
```

Comment:
```suggestion
enum class SchedulingMode {
  Manual,
  InterDeviceOnly,
  ReductionScheduler,
  Automatic,
};
```
I'd like to sort them from most manual to most automatic. Correct me if the order is wrong.

Also, I'd add verbose comments on what each means, especially now that we have some options between fully manual and fully automatic.

Reply:
The way it was sorted before was the right order. I've added some comments!